sim/midas/src/main/scala/midas/models/dram/IngressUnit.scala

```diff
@@ -75,12 +78,21 @@ class IngressModule(val cfg: BaseConfig)(implicit val p: Parameters) extends Mod
   awCredits.inc := wQueue.io.enq.fire() && wQueue.io.enq.bits.last
   awCredits.dec := awQueue.io.deq.fire()
 
+  val addrWidth = p(NastiKey).addrBits
+  val idWidthBits = new NastiWriteDataChannel 
+  val idWidth = idWidthBits.id   
+
+  val arIDQueue = Module(new Queue(idWidth, cfg.maxReads*2))
+  val awIDQueue = Module(new Queue(idWidth, cfg.maxWrites*2))
+
   // All the sources of host stalls
   val tFireHelper = DecoupledHelper(
     io.nastiInputs.hValid,
     awQueue.io.enq.ready,
     wQueue.io.enq.ready,
-    arQueue.io.enq.ready)
+    arQueue.io.enq.ready,
```

Comment:
If these are sized large enough (2x the size), you might not need to include them here (i'd put assertions on the readies though, to ensure they never assert).

If you're going to use the minimally sized structures, which is probably adviseable, you'll probably need to include the FIFO address matchers in here too, no?

Reply:
The dequeue conditions are same as arIDQueue and awIDQueue for réad FIFO Address matcher and write FIFO Address Matcher respective. arIDQueue and awIDQueue have been included in the List.